inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -445,21 +445,61 @@ public KBHandle createConcept(KnowledgeBase kb, KBConcept aConcept)
         });
     }
     
-    @Override
-    public Optional<KBConcept> readConcept(KnowledgeBase kb, String aIdentifier)
+    @Override 
+    public Optional<KBConcept> readConcept(KnowledgeBase aKB, String aIdentifier)
         throws QueryEvaluationException
     {
-        return read(kb, (conn) -> {
-            ValueFactory vf = conn.getValueFactory();
-            Resource subject = vf.createIRI(aIdentifier);
-            if (RdfUtils.existsStatementsWithSubject(conn, subject, false)) {
-                KBConcept kbConcept = KBConcept.read(conn, vf.createIRI(aIdentifier),kb);
-                return Optional.of(kbConcept);
-            }
-            else {
-                return Optional.empty();
+        KBHandle resultHandle = read(aKB, (conn) -> {
+            String QUERY = SPARQLQueryStore.readConcept(aKB, 1);
+            ValueFactory vf = SimpleValueFactory.getInstance();
+            TupleQuery tupleQuery = conn.prepareTupleQuery(QueryLanguage.SPARQL, QUERY);
+            tupleQuery.setBinding("oItem", vf.createIRI(aIdentifier));
+            tupleQuery.setBinding("pTYPE", aKB.getTypeIri());
+            tupleQuery.setBinding("oCLASS", aKB.getClassIri());
+            tupleQuery.setBinding("pSUBCLASS", aKB.getSubclassIri());
+            tupleQuery.setBinding("pLABEL", aKB.getLabelIri());
+            tupleQuery.setBinding("pDESCRIPTION", aKB.getDescriptionIri());
+            tupleQuery.setIncludeInferred(false);
+            TupleQueryResult result = tupleQuery.evaluate();
+
+            KBHandle handle = new KBHandle();
+            while (result.hasNext()) {
+                BindingSet bindings = result.next();
+                if (bindings.size() == 0) {
+                    continue;
+                }
+                Binding label = bindings.getBinding("l");
+                Binding description = bindings.getBinding("d");
+                handle.setIdentifier(aIdentifier);
+                if (label != null) {
+                    handle.setName(label.getValue().stringValue());
+                    if (label.getValue() instanceof Literal) {
+                        Literal literal = (Literal) label.getValue();
+                        Optional<String> language = literal.getLanguage();
+                        language.ifPresent(handle::setLanguage);
```

Comment:
Can we move this part to ``evaluateListQuery(...)`` and then call ``evaluateListQuery(...)`` inside ``readConcept(...)`` to avoid redundant code?